lib/util/regex.ts

```diff
@@ -0,0 +1,22 @@
+import { logger } from '../logger';
+
+export interface RegExConstructor {
+  new (pattern: RegExp | string): RegExp;
+  new (pattern: string, flags?: string): RegExp;
+}
+
+const RegEx: RegExConstructor = (() => {
+  try {
+    // eslint-disable-next-line
+    const RE2 = require('re2');
+    // Test if native is working
+    new RE2('.*').exec('test');
+    logger.debug('Using RE2 as regex engine');
```

Comment:
Will this debug log 1 time per run or N times?

Reply:
Should be on per run.